bench/irmin-pack/tree.ml

```diff
@@ -532,14 +142,32 @@ module Bench_suite (Store : Store) = struct
         config.ncommits config.nchain_trees config.depth repo_pp
         Benchmark.pp_results result
 
-  let run_read_trace = Trace_replay.run
+  let run_read_trace config =
+    let replay_config : Irmin_traces.Trace_replay.config =
+      {
+        ncommits_trace = config.ncommits_trace;
+        store_dir = config.store_dir;
+        path_conversion = config.path_conversion;
+        inode_config = config.inode_config;
+        store_type = config.store_type;
+        commit_data_file = config.commit_data_file;
+        artefacts_dir = config.artefacts_dir;
+        keep_store = config.keep_store;
+        keep_stat_trace = config.keep_stat_trace;
+        no_summary = config.no_summary;
+        empty_blobs = config.empty_blobs;
+      }
```

Comment:
This feels off; why do we want both an abstract and a concrete config to convert between?

Reply:
3 types of config here:
1. `Tree.config` is the mirror of the `tree.exe` CLI parameters
2. `Trace_replay.config` is specific to the trace replay
3. `Store.store_config` is the config to instantiate a repo.

Having both `1.` and `2.` seem unavoidable to me.

Having `3` is annoying, but I didn't find a better way.

The benchmarks are parameterised by the type of store but are also in charge of opening and closing the repo. This means that the configuration for opening the repo is abstract, when seen from the call to `create_repo` (inside `run`), it must be provided to `run`. Hence the config `3.`.

Reply:
Makes sense, thanks. This pattern suggests to me we might want to switch to using objects at some point, but regardless it's not important now.